### DISCUSSION TITLE
Shelly Flood water message is not signalled in HomeKit.

### DISCUSSION THREAD

**Discussion by hatty24** [Luligu/matterbridge-shelly#96] [Q&A]
Hello,

I have several Shelly Plus 1PM and Plus 2PM in use, all of which are controlled via the Matterbridge with HomeKit. That works perfectly!

I also have a Shelly Flood. Although the temperature of the Flood is displayed correctly in HomeKit, but the most important thing, the water message, is not transmitted.
Regardless of the status, in HomeKit the contact is permanently displayed as closed.

In any case, the Shelly Flood works perfectly. The sensor status is displayed correctly both in the Shelly app and via Homebridge. Only not via Matterbridge.

Both Matterbridge and the Shelly plugin are up to date.

Does anyone have any idea what the problem is?
I would be very grateful for your help.

**Comment by Luligu** [comment]
Hi,
I checked just to be sure.
All is correct.

@tammeryousef1006 can you please check yours?

Do you have any non standard configuration?

![image](https://github.com/user-attachments/assets/d59ac184-36c7-43d6-bdb2-813518ffcc2a)

**Reply by tammeryousef1006** [reply]
just tested now , its working well when leak detected and reflect correctly to controller 

![Screenshot 2024-12-02 223521](https://github.com/user-attachments/assets/514e3093-23f2-47fe-a13e-b4b3c7220761)
![Screenshot 2024-12-02 223414](https://github.com/user-attachments/assets/3c04daa5-7b9a-4ea4-9965-2e30dac1b00a)

**Comment by tammeryousef1006** [comment]
![Screenshot 2024-12-02 223630](https://github.com/user-attachments/assets/2e2aeb74-7dc3-4bc3-a96f-eeb75bba2b87)

check please if CoIoT peer enabled and set to mcast

**Comment by hatty24** [comment]
@Luligu , @tammeryousef1006 

Many thanks for your support.

The Matterbridge is running via Docker on a Synology DS218+ with DSM7.

CoIoT peer is enabled and set to mcast.

![IMG_0308](https://github.com/user-attachments/assets/2f410273-fa0d-4c2f-9673-9d5de8eccf15)

I have reset the Shelly Flood to factory settings and set it up again from scratch.
Now all messages are coming through correctly, for whatever reason.

![IMG_0309](https://github.com/user-attachments/assets/dd810652-9a56-4b55-9dcb-c15d57cb894e)

Thanks again for your quick answers.

**Reply by tammeryousef1006** [reply]
glad to hear that and thanks for feedback
